internal/configsource/envvarconfigsource/README.md

```diff
@@ -0,0 +1,44 @@
+# Environment Variable Config Source (Alpha)
+
+Use the environmental variable config source instead of direct references to
+environment variables in the config to inject YAML fragments or to have default
+values in case the selected environment variable is undefined.
+
+## Configuration
+
+Under the `config_sources:` use `env:` or `env/<name>:` to create an
+environment variable config source. The following parameters are available to customize
+environment variable config sources:
+
+```yaml
+config_sources:
+  env:
+    # defaults is used to create a set of fallbacks in case the original env var is
+    # undefined in the environment.
+    defaults:
+      MY_ENV_VAR: my env var value
+```
+
+It is possible to fail the configuration load if an environment variable is required
+and must be defined either on the environment or on the `defaults` of the config source.
```

Comment:
Do you know which case is more common? I don't have any data but I feel it might  be more common to expect an env var to be defined. If so, may be that can be the default behavior and we can add `?optional=true` param instead.